### PR TITLE
KG - Reports Refactors

### PR DESCRIPTION
**Pull request by kyle-glick**

### CODE REVIEW

File: app/assets/javascripts/documents.js.coffee [hunk 1]
```diff
@@ -18,11 +18,21 @@ $ ->
 
     $(document).on 'click', 'a.attached_file', ->
       update_view_on_download_new_report $(this), 'table.documents', 'Identity'
-      
+
+    $(document).on 'click', 'a.remove-document', ->
+      document_id = $(this).attr('document_id')
```
Comment (goleksiak) [inline app/assets/javascripts/documents.js.coffee]: This should be stored as a data attribute

File: app/helpers/documents_helper.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module DocumentsHelper
 
-  def format_date date
+  def format_document_date date
```
Comment (goleksiak) [inline app/helpers/documents_helper.rb]: Use parenthesis around parameters in function definitions: https://github.com/bbatsov/ruby-style-guide#method-parens

File: app/helpers/documents_helper.rb [hunk 3]
```diff
@@ -26,4 +26,15 @@ def edit_formatter document
...
   end
 
+  def delete_formatter document
+    if document.completed?
+    [
+      "<a class='remove remove-document' href='javascript:void(0)' title='Remove' document_id='#{document.id}'>",
+      "<i class='glyphicon glyphicon-remove'></i>",
+      "</a>"
+    ].join ""
+  else
```
Comment (goleksiak) [inline app/helpers/documents_helper.rb]: Indention issues

File: app/assets/javascripts/documents.js.coffee [hunk 4]
```diff
@@ -18,11 +18,21 @@ $ ->
...
+      del = confirm "Are you sure you want to delete this document?"
+      if del
+        if $(this).parent().siblings("td.viewed_at").text() == ""
+          add_to_report_notification_count('Identity', -1)
+        $.ajax
+          type: 'DELETE'
+          url: "/documents/#{document_id}"
```
Comment (goleksiak) [inline app/assets/javascripts/documents.js.coffee]: Specify format in URLs

**Comment by kyle-glick** [comment]
This pull request is ready for review

**Comment by Stuart-Johnson** [comment]
This looks good to me.

File: app/controllers/documents_controller.rb [hunk 5]
```diff
@@ -74,6 +78,18 @@ def update
     end
   end
 
+  def destroy
+    respond_to do |format|
+      format.js {
+        @document.destroy
+        
+        mark_document_as_accessed
```
Comment (kyle-glick) [inline app/controllers/documents_controller.rb]: Look into this

**Pull request merged**